### PR TITLE
bug and issue fixes for 3.0.1

### PR DESCRIPTION
**Pull request by yarneo**
* Try to remove the redundant error for module definition of Mixpanel in HelloMixpanel app
* Fix problematic and missing header files in Mixpanel Targets
* Connect showNetworkActivityIndiciator and useIPAddressForGeoLocation so they will work again
* Fix XCode8/iOS10 error of warn_unused_result for SecRandomCopyBytes
* Rename the parameter in charge of enabling connection to visual display for A/B Test and codeless
* Remove Macro for A/B test
* Remove Deprecated NSURLConenction and replace with NSURLSession

### CODE REVIEW

File: Mixpanel/Mixpanel.m [hunk 1]
```diff
@@ -154,6 +154,28 @@ - (void)setValidationEnabled:(BOOL)validationEnabled {
 }
 #endif
 
+- (BOOL)showNetworkActivityIndicator {
+    return _showNetworkActivityIndicator;
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Why even have a bool here? Let's pass along the one from MPNetwork instead

File: Mixpanel/Mixpanel.m [hunk 2]
```diff
@@ -154,6 +154,28 @@ - (void)setValidationEnabled:(BOOL)validationEnabled {
...
+}
+
+- (void)setShowNetworkActivityIndicator:(BOOL)showNetworkActivityIndicator {
+    _showNetworkActivityIndicator = showNetworkActivityIndicator;
+    if (self.network) {
+        self.network.showNetworkActivityIndicator = showNetworkActivityIndicator;
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: We don't need if statements like this in Objective C

File: Mixpanel/Mixpanel.m [hunk 3]
```diff
@@ -728,9 +750,13 @@ - (NSString *)IFA
     if (ASIdentifierManagerClass) {
         SEL sharedManagerSelector = NSSelectorFromString(@"sharedManager");
         id sharedManager = ((id (*)(id, SEL))[ASIdentifierManagerClass methodForSelector:sharedManagerSelector])(ASIdentifierManagerClass, sharedManagerSelector);
-        SEL advertisingIdentifierSelector = NSSelectorFromString(@"advertisingIdentifier");
-        NSUUID *uuid = ((NSUUID* (*)(id, SEL))[sharedManager methodForSelector:advertisingIdentifierSelector])(sharedManager, advertisingIdentifierSelector);
-        ifa = [uuid UUIDString];
+        SEL advertisingTrackingEnabledSelector = NSSelectorFromString(@"isAdvertisingTrackingEnabled");
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Won't this break IDFA on iOS 7?

File: Mixpanel/Mixpanel.m [hunk 4]
```diff
@@ -871,7 +897,14 @@ - (void)setUpListeners
                                name:@"com.parse.bolts.measurement_event"
                              object:nil];
 
-#if !defined(DISABLE_MIXPANEL_AB_DESIGNER) && !defined(MIXPANEL_TVOS_EXTENSION)
+    if (self.enableVisualABTestAndCodeless) {
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: This will always be true

File: Mixpanel/MPNetwork.m [hunk 5]
```diff
@@ -279,7 +279,7 @@ + (BOOL)parseHTTPFailure:(NSHTTPURLResponse *)response withError:(NSError *)erro
 
 - (void)updateNetworkActivityIndicator:(BOOL)enabled {
 #if !MIXPANEL_LIMITED_SUPPORT
-    if (self.shouldManageNetworkActivityIndicator) {
+    if (self.showNetworkActivityIndicator) {
```
Comment (yarneo) [inline Mixpanel/MPNetwork.m]: Was using the same variable that remained on Mixpanel.h and was part of the public API. I have no preference on the naming.

File: Mixpanel/Mixpanel.m (hunk 3 above)
Comment (yarneo) [inline Mixpanel/Mixpanel.m]: we don't support ios7 anymore

File: Mixpanel/Mixpanel.m [hunk 6]
```diff
@@ -1611,7 +1651,7 @@ - (void)connectToABTestDesigner
 - (void)connectToABTestDesigner:(BOOL)reconnect
 {
     // Ignore the gesture if the AB test designer is disabled.
-    if (!self.enableABTestDesigner) return;
+    if (!self.enableVisualABTestAndCodeless) return;
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Good defensive coding

File: Mixpanel/Mixpanel.m [hunk 7]
```diff
@@ -154,6 +154,25 @@ - (void)setValidationEnabled:(BOOL)validationEnabled {
...
+- (BOOL)shouldManageNetworkActivityIndicator {
+    return self.network.shouldManageNetworkActivityIndicator;
+}
+
+- (void)setShouldManageNetworkActivityIndicator:(BOOL)shouldManageNetworkActivityIndicator {
+    self.network.shouldManageNetworkActivityIndicator = shouldManageNetworkActivityIndicator;
+}
+
+- (BOOL)useIPAddressForGeoLocation {
+    return _useIPAddressForGeoLocation;
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Do we need this BOOL?

Comment (yarneo) [inline Mixpanel/Mixpanel.m]: we use it for the tracking integration method :|

Comment (yarneo) [inline Mixpanel/Mixpanel.m]: I could move the entirety of that to MPNetwork and get rid of it. lemme know if that sounds good

File: Mixpanel/Mixpanel.m (hunk 3 above)
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Touché

**Comment by samgreen** [comment]
Fix the last BOOL and LGTM

File: Mixpanel/Mixpanel.h [hunk 8]
```diff
@@ -230,14 +230,14 @@ NS_ASSUME_NONNULL_BEGIN
...
- tests and their variants will still be delivered.
+ Controls whether to enable the visual test designer for A/B testing and codeless on mixpanel.com. 
+ You will be unable to edit A/B tests and codeless events with this disabled, however *previously*
+ created A/B tests and codeless events will still be delivered.
  
  @discussion
  Defaults to YES.
  */
-@property (atomic) BOOL enableABTestDesigner;
+@property (atomic) BOOL enableVisualABTestAndCodeless;
```
Comment (mdiep) [inline Mixpanel/Mixpanel.h]: Renaming these properties was a breaking change from the 3.0.0 version. 😡

**Pull request merged**